packages/angular_devkit/schematics/src/tree/interface.ts

```diff
@@ -91,6 +91,16 @@ export interface Tree {
    * @returns A string containing the contents of the file; if not found, returns `null`.
    */
   readText(path: string): string | null;
+
+  /**
+   * Reads and parses a file from the Tree as a UTF-8 encoded JSON file.
+   * Parsing supports JSON comments and trailing commas.
+   *
+   * @param path The path of the file to read.
+   * @returns A JsonValue containing the parsed contents of the file; if not found, returns `undefined`.
+   */
+  readJSON(path: string): JsonValue | undefined;
```

Comment:
Nit: Do we have any standard casing convention for JSON? Should this be `readJson`?

I see that `JsonValue` is lower cased, but I'm not sure if we have any public APIs which use the term JSON and should make sure we're consistent with them.